File: androidApp/src/main/java/se/fransman/tgapp/composables/screens/HomeScreen.kt
package se.fransman.tgapp.composables.screens

import androidx.compose.foundation.layout.*
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import se.fransman.tg.shared.viewmodel.screens.home.HomeState

@Composable
fun HomeScreen(
  homeState: HomeState,
) {

  if (homeState.isLoading) {

    LoadingScreen()

  } else {
    val data = homeState.players
    Column(modifier = Modifier.padding(10.dp)) {
      for (player in data) {
        Text(text = "${player.firstName} ${player.lastName}", style = MaterialTheme.typography.body1)
      }
    }

  }
}

File: iosApp/iosApp/Views/screens/home/HomeScreen.swift
import SwiftUI
import shared

struct HomeScreen: View {
    var homeState: HomeState
    
    var body: some View {
        VStack {
            if homeState.isLoading {
                LoadingScreen()
            } else {
                let data = homeState.players
                ScrollView(.vertical) {
                    VStack(alignment: .leading, spacing: 5) {
                        ForEach(data, id: \.id) { player in
                            Text("\(player.firstName) \(player.lastName)").font(Font.callout)
                        }
                    }.padding(.all, 15).frame(maxWidth: .infinity, alignment: .leading)
                }
            }
        }
    }
}
